akka-actor/src/main/scala/akka/io/Tcp.scala

```diff
@@ -425,7 +425,7 @@ object Tcp extends ExtensionId[TcpExt] with ExtensionIdProvider {
    * Whenever a command cannot be completed, the queried actor will reply with
    * this message, wrapping the original command which failed.
    */
-  case class CommandFailed(cmd: Command) extends Event
+  case class CommandFailed(cmd: Command, cause: Option[Throwable] = None) extends Event
```

Comment:
Unfortunately it basically impossible to extend a case class in a binary compatible fashion (meaning: I cannot prove its impossibility but I have never seen a working proposal). Therefore we cannot do this in the 2.3.x series (which current Streams work is based upon).